mistral-search-engine.ipynb

```diff
@@ -0,0 +1,721 @@
+       "pydevconsole        NA\n",
+       "pydevd              2.9.5\n",
+       "pydevd_file_utils   NA\n",
+       "pydevd_plugins      NA\n",
+       "pydevd_tracing      NA\n",
+       "pygments            2.18.0\n",
+       "pytz                2024.1\n",
+       "six                 1.16.0\n",
+       "sniffio             1.3.1\n",
+       "soupsieve           2.5\n",
+       "stack_data          0.6.2\n",
+       "swig_runtime_data4  NA\n",
+       "tornado             6.3.3\n",
+       "traitlets           5.14.3\n",
+       "typing_extensions   NA\n",
+       "urllib3             2.2.1\n",
+       "vscode              NA\n",
+       "wcwidth             0.2.13\n",
+       "yarl                1.9.4\n",
+       "zmq                 25.1.2\n",
+       "zoneinfo            NA\n",
+       "</pre>\n",
+       "</details> <!-- seems like this ends pre, so might as well be explicit -->\n",
+       "<pre>\n",
+       "-----\n",
+       "IPython             8.25.0\n",
+       "jupyter_client      8.6.2\n",
+       "jupyter_core        5.5.0\n",
+       "-----\n",
+       "Python 3.10.14 (main, May  6 2024, 14:42:37) [Clang 14.0.6 ]\n",
+       "macOS-14.4-arm64-arm-64bit\n",
+       "-----\n",
+       "Session information updated at 2024-06-11 03:48\n",
+       "</pre>\n",
+       "</details>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "execution_count": 81,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import session_info\n",
+    "session_info.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install aiohttp bs4 faiss-cpu lxml mistralai nest_asyncio nest_asyncio pandas requests python-dotenv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 80,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dotenv import load_dotenv\n",
+    "import os\n",
+    "\n",
+    "load_dotenv()  # load environment variables from .env file\n",
+    "MISTRAL_API_KEY = os.getenv(\"MISTRAL_API_KEY\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scraper Definitions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Fetching page: 1\n",
+      "Fetching page: 2\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Exception ignored in: <function MistralClient.__del__ at 0x11efd1ab0>\n",
+      "Traceback (most recent call last):\n",
+      "  File \"/Users/hugolebelzic/miniconda3/envs/mistral-cookbook-contrib/lib/python3.10/site-packages/mistralai/client.py\", line 49, in __del__\n",
+      "    self._client.close()\n",
+      "AttributeError: 'MistralClient' object has no attribute '_client'\n"
```

Comment:
Is it possible to handle this error?

Reply:
Hello, I ran it again and wasn't able to reproduce this error. Do you have more indications ?

Reply:
It seems the error was fixed by itself, weirdly...
Btw, if possible keep the !pip install for the rest of the required libraries!
Otherwise I think everything is alright.

Reply:
just added it! everything should be good now